packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -1,9 +1,14 @@
 import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useMutation, useQuery } from '@apollo/client';
+import { useIntl } from 'react-intl';
 import { useCartContext } from '@magento/peregrine/lib/context/cart';
-import { deriveErrorMessage } from '../../../util/deriveErrorMessage';
+import { useUserContext } from '@magento/peregrine/lib/context/user';
 import configuredVariant from '@magento/peregrine/lib/util/configuredVariant';
+
+import { useWishlist } from './useWishlist';
```

Comment:
I feel like having this hook live as a sibling to the talon within the `talons` directory could be confusing. I know we spoke about this regarding @tjwiebell's use case. Did we end up with a decision on where to place these utility hooks?

Reply:
Not quite completely though. I am trying to see if I can generalize wishlist logic into a talon of its own so all the components/talons that use it can just import. But unfortunately, that can be tricky because the use cases can be different and Jimmy suggested moving over it if it's getting complex to generalize.

Reply:
Yea I agree. At a minimum, throw the file into a `helpers` directory.

Reply:
If I am able to generalize this, yes I will move it into a helper's folder but if it is bound to stay like this, it's not a helper anymore so we can leave it as it is. I'll update soon about the generalization work and let's make the call then.